careers/software-engineer.md

```diff
@@ -4,52 +4,56 @@ The goal of the Software Engineer path is to train engineers to master the funda
 
-**Technical Scope:** An Engineer I is capable of taking small well-scoped components of larger projects and features and, with the mentoring and assistance of more senior engineers, completing these tasks in a reasonable time frame. Engineers at this level work with more senior members of the Engineering team to ensure they make steady progress on tasks that are assigned to them. More senior engineers may also help when they are blocked.
+**Technical Knowledge:** They have a theoretical understanding of the core concepts and best practices of software engineering and have some practical experience implementing them. They understand the basics of writing performant, well-abstracted code, but need help with complex tasks and tasks outside their specialization.
 
-Engineers at this level will have input into the estimation and planning process, and will collaborate with more senior resources to scope out work and understand the larger risks and ramifications of the changes they are making.
+**Productivity:** They are capable of completing small, well-defined tasks on time with minimal help from other engineers. They are mindful of their technical limitations and take them into account when planning their work. They focus on reliably delivering quality work on time.
 
-**Communication:** They are learning how to communicate well and how to deliver feedback to peers and their manager. When given a task with unclear requirements they are learning how to ask for clarification and how to identify underlying assumptions in design and implementation and how to clarify them. They understand how their work fits into the larger picture for their team, and can work with product and other engineers to help identify conflicting requirements. They will start to participate more in the technical design process, often with guidance from more senior engineers.
+**Impact:** They take on work that helps the team meet its sprint goals. They actively participate in team norms, processes, and expectations. They focus on delivering the work assigned to them in a sprint.
 
-By the time an Engineer I is ready to be promoted to Engineer II, they will have expertise in Engineering best practice, tools, techniques and a solid introduction to the technologies in their domain.
+**Communication:** They ask questions to understand how their work fits into the larger context. They proactively communicate whether their work is ahead, on track, or behind, and are especially mindful about communicating when work is taking longer than expected. They welcome feedback and apply the feedback they received in other contexts.
 
-## Engineer II
+**Collaboration:** They pair frequently. They are able to quickly recognize when they are stuck and ask for help. For more complicated tasks, they leverage the knowledge of their peers to plan and execute their work on time. They actively participate in code review and offer feedback to others.
```

Comment:
Do Apprentices not get assigned PRs to review? It makes sense if that's the case, I just don't know if we have a mechanism in the auto-assigner to easily filter out developers at that level. 

FWIW, Aidan has been reviewing PRs and has done a pretty decent job so far.

Reply:
I do expect apprentices to participate in code review. However, due to their unconscious incompetencies, I wouldn't expect the reviews to be particularly insightful unless they pair with someone more senior.

In contrast, at EN1, I'd expect them to have the basic skills to offer constructive feedback -- or have the self-awareness to know where they need help reviewing.

I think your point about Aidan indicates that in this area they are performing beyond the level of apprentice.

Reply:
Maybe we could rephrase that last sentence to something like `They provide constructive feedback in code review and offer helpful feedback to others`?

Reply:
Modified it slightly to "They provide constructive feedback in code review and know when they need help constructing meaningful feedback."